--- src/Model/ApiResourceBase.php	2021-10-20 20:22:39.000000000 +0000
+++ src/Model/ApiResourceBase.php	2023-12-14 09:39:41.669081360 +0000
@@ -66,7 +66,7 @@
     /**
      * @inheritdoc
      */
-    public function offsetExists($offset)
+    public function offsetExists(mixed $offset): bool
     {
         return $this->hasProperty($offset);
     }
@@ -97,7 +97,7 @@
     /**
      * @inheritdoc
      */
-    public function offsetGet($offset)
+    public function offsetGet(mixed $offset): mixed
     {
         return $this->getProperty($offset, false);
     }
@@ -120,7 +120,7 @@
      *
      * @throws \BadMethodCallException
      */
-    public function offsetSet($offset, $value)
+    public function offsetSet($offset, $value): void
     {
         throw new \BadMethodCallException('Properties are read-only');
     }
@@ -130,7 +130,7 @@
      *
      * @throws \BadMethodCallException
      */
-    public function offsetUnset($offset)
+    public function offsetUnset(mixed $offset): void
     {
         throw new \BadMethodCallException('Properties are read-only');
     }
